crates/matrix-sdk-crypto/src/olm/session.rs

```diff
@@ -285,3 +275,70 @@ pub struct PickledSession {
     /// The Unix timestamp when the session was last used.
     pub last_use_time: SecondsSinceUnixEpoch,
 }
+
+#[cfg(test)]
+mod tests {
+    use matrix_sdk_test::async_test;
+
+    use ruma::{device_id, user_id};
+    use serde_json::{self, Value};
+    use vodozemac::olm::{OlmMessage, SessionConfig};
+
+    use crate::{
+        identities::ReadOnlyDevice, olm::Account,
+        types::events::room::encrypted::ToDeviceEncryptedEventContent,
+    };
+
+    #[async_test]
+    async fn test_encryption_and_decryption() {
```

Comment:
I'm finding it hard to work out what this test does. Could it have a more specific name, and/or could you add some comments? I often add "given, when, then" comments to explain what I'm doing, but here what I'm really looking for is which line of code is the actual thing we're testing?

Reply:
Is it "Given some stuff, when we create an inbound session, then it contains the signed device keys"?